refinery/data_set_manager/tasks.py

```diff
@@ -273,22 +277,26 @@ def parse_isatab(username, public, path, identity_id=None,
         return data_set_uuid
 
 
-@task()
-def generate_auxiliary_file(auxiliary_node, parent_node_file_store_item):
+@task(soft_time_limit=3600)
+def generate_auxiliary_file(parent_node_uuid):
     """Task that will generate an auxiliary file for visualization purposes
     with specific file generation tasks going on for different FileTypes
     flagged as: `used_for_visualization`.
-    :param auxiliary_node: a Node instance
-    :type auxiliary_node: Node
-    :param datafile_path: relative path to datafile used to generate aux file
-    :type datafile_path: String
-    :param parent_node_file_store_item: FileStoreItem associated with the
-    parent Node
-    :type parent_node_file_store_item: FileStoreItem
+    :param parent_node: the parent Node uuid
+    :type parent_node_file_store_item: Node
     """
     generate_auxiliary_file.update_state(state=celery.states.STARTED)
+    parent_node = Node.objects.get(uuid=parent_node_uuid)
+    datafile = parent_node.file_item.datafile
+    auxiliary_file_store_item = FileStoreItem.objects.create()
+    auxiliary_node = parent_node.create_and_associate_auxiliary_node(
+        auxiliary_file_store_item
+    )
     try:
-        datafile_path = parent_node_file_store_item.datafile.path
+        if not settings.REFINERY_S3_USER_DATA:
+            datafile_path = datafile.path
+        else:
+            datafile_path = datafile.name
     except (NotImplementedError, ValueError):
         datafile_path = None
     try:
```

Comment:
This try block is huge. I think the only function that can raise exceptions is `generate_bam_index()`.